task/rhacs-m2m-authenticate/4.4.2/rhacs-m2m-exchange.yaml

```diff
@@ -0,0 +1,103 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: rhacs-m2m-exchange
+  labels:
+    apps.kubernetes.io/version: "4.5"
```

Comment:
a) this doesn't match the version path in the directory b) I don't think we should version it similar to the ACS version, but keep it as a separate entity. Otherwise things will get a bit awkward if we don't release a new version with each release of ACS version (I'd think as a user "why hasn't this been updated to the latest ACS version?").

Reply:
Not sure I agree with the `10.0` but rather would name it `1.0`. While there's likelihood that in the future things may overlap it's quite hard to find arguments for the random `10.0` version and would be quite confusing as a user; whereas if you'd look at the version history starting from `1.0` it would be somewhat clear that this follows a different version scheme than Central _even if _ we overlap with the ACS version eventually.

Reply:
We could deliver the rhacs-m2m-exchange task as version 1.0 or 0.1 . Now the rhacs-deployment-check, rhacs-image-check and rhacs-image-scan are already delivered as 3.71. If we want a consistent versioning across tasks, these modified tasks would be delivered with version 1.0 too, but using 1.0 would suggest the 3.71 versions are "newer" whereas they are not. I think this would be misleading in the end, hence my suggestion to have 10.0 as new version.

Reply:
So for the new M2M task we do not have to have parity with the other tasks versions so I would say this should be `1.0` or `0.1`, both work for me.

For the other tasks "new" version: I still think it should be `4.0` for those, as a jump to `10.0` would be more confusing as to what happened between "3.X - 10.0" especially since we cannot really give a good explanation. We should just follow basic SemVer logic here and bump the major and be done with, even though it does unfortunately create the parallel to the ACS major version, but doesn't seem like we can _properly_ fix it without artificially creating a big version jump (which I don't see making things clearer for the user).